src/components/Card.js

```diff
@@ -1,57 +1,87 @@
 export default class Card {
-  constructor(cardData, templateSelector, handleCardClick) {
+  constructor(
+    { cardData, myId, handleImageClick, handleDeleteClick, handleLikeClick },
+    cardSelector
+  ) {
     this._cardData = cardData;
-    this._link = cardData.link;
+    this._myId = myId;
     this._name = cardData.name;
-    this._templateSelector = templateSelector;
-    this._handleCardClick = handleCardClick;
+    this._link = cardData.link;
+    this._cardSelector = cardSelector;
+    this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
+    this._likes = cardData.likes;
+    this._owner = cardData.owner;
+    this._id = cardData._id;
   }
-  _setEventListeners() {
-    this._likeButton = this._cardElement.querySelector(".card__like-button");
-    this._deleteButton = this._cardElement.querySelector(".card__delete");
 
-    this._likeButton.addEventListener("click", () => this._toggleLike());
+  _setEventListeners() {
+    this._likeButton = this._elment.querySelector(".card__like-button");
+    this._deleteButton = this._elment.querySelector(".card__delete-button");
+    this._likesAmount = this._elment.querySelector(".card__like-amount");
+    this._likeButton.addEventListener("click", () => this._handleLikeClick());
     this._deleteButton.addEventListener("click", () =>
-      this._handleDeleteButton()
+      this._handleDeleteClick()
     );
 
-    this._cardImageEl.addEventListener("click", () =>
-      this._handleCardClick(this._cardImageEl)
-    );
+    this._cardImage.addEventListener("click", () => {
+      this._handleImageClick(this._cardData);
+    });
+  }
+  getId() {
+    return this._id;
+  }
+  isLiked() {
+    return this._likes.some((like) => like._id === this._myId);
+  }
+  setLikes(likes) {
+    this._likes = likes;
+    this._handleLikeIcon();
+    this.updateLikes();
+  }
+  updateLikes() {
+    this._likesAmount.textContent = this._likes.length;
   }
 
-  _toggleLike = () => {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button-active");
-  };
+  _handleLikeIcon() {
+    if (this.isLiked()) {
+      this._likeButton.classList.add("card__like-button_active");
+    } else {
+      this._likeButton.classList.remove("card__like-button_active");
+    }
+  }
+  _checkCardOwner() {
```

Comment:
Method name should describe what the method does. This one hides delete button, so the good name would be `_hideTrashButton`